open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -190,6 +209,142 @@ def create_docker_host_vm(self, hackathon_id):
         self.sche.add_once('docker_host_manager', 'check_vm_status', context=context, minutes=5)
         return True
 
+    def create_host_server(self, hackathon_id, args):
+        """
+        create a docker host DB object for a hackathon and insert record into the database.
+        param-"args" contain all necessary infos to new a docker_host
+
+        :param hackathon_id: the id of hackathon in DB
+        :type hackathon_id: Integer
+
+        :return: return True if succeed, otherwise return False
+        :rtype: bool
+        """
+
+        try:
+            ping_url = 'http://%s:%d/_ping' % (args.public_dns, int(args.public_docker_api_port))
```

Comment:
It's better to separate the `_ping` login into a separate method  in `hosted_docker`. I remember there is already one in `hosted_docker`? please check that. I don't think it's a good manner to call docker API in other classes than `hosted_docker`. All direct talk with docker should be enclosed into a single place.